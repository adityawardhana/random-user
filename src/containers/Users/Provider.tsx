import React, { FC, useEffect, useState } from "react";
import qs from "query-string";
import { getUsers } from "../../services/users";
import useFetch from "../../hooks/useFetch";
import useDebounce from "../../hooks/useDebounce";
import UserContext from "./Context";

const Provider: FC<{}> = ({ children }) => {
  const [params, setParams] = useState({
    inc: "gender,name,login,email,registered",
    results: 10,
    page: 1,
    keyword: "",
    sortBy: "",
    sortOrder: "",
    gender: "",
  });
  const [keyword, setKeyword] = useState("");
  const [searchFocus, setSearchFocus] = useState(false);

  const {
    fetch,
    loading,
    error,
    data: { results: users } = {},
  } = useFetch(
    (newParams) =>
      getUsers(qs.stringify(newParams || params, { skipEmptyString: true })),
    { immediate: false }
  );

  const handleChangeSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
    setKeyword(e.target.value);
  };

  const handleSelectGender = (value: string) => {
    setParams({ ...params, gender: value });
  };

  const handleChangeSort = (sortBy: string, sortOrder: string) => {
    setParams({ ...params, sortBy, sortOrder, page: 1 });
  };

  const handleChangePage = (page: number) => {
    setParams({ ...params, page });
  };

  const handleResetFilter = () => {
    setKeyword("");
    setParams({ ...params, page: 1, keyword: "", gender: "" });
  };

  useDebounce(
    () => {
      searchFocus && setParams({ ...params, keyword, page: 1 });
    },
    keyword,
    1000
  );

  useEffect(() => {
    fetch({ ...params });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [params]);

  return (
    <UserContext.Provider
      value={{
        params,
        users,
        loading,
        error,
        keyword,
        setSearchFocus,
        handleChangePage,
        handleChangeSearch,
        handleChangeSort,
        handleSelectGender,
        handleResetFilter,
      }}
    >
      {children}
    </UserContext.Provider>
  );
};

export default Provider;
