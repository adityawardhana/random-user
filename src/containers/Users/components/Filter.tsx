import { Box, Button } from "@adityawardhana/ts-react-ui"
import Dropdown from "../../../ui/Dropdown";
import SearchBox from "../../../ui/SearchBox";
import { Gender } from "../Constant";
import useUsers from "../useUser";

const Filter = () => {
  const {
    keyword,
    handleChangeSearch,
    setSearchFocus,
    handleSelectGender,
    handleResetFilter,
    params,
  } = useUsers();
  return (
    <Box mb={40} alignItems="center">
      <Box width={200} mr={16}>
        <SearchBox
          placeholder="Type your search here..."
          value={keyword}
          onChange={handleChangeSearch}
          onFocus={() => setSearchFocus(true)}
          onBlur={() => setSearchFocus(false)}
        />
      </Box>
      <Box width={200} mr={16}>
        <Dropdown
          label={Gender.find((key) => key.value === params.gender)?.label || ""}
          content={Gender}
          onChange={(data) => {
            if (data.value !== params.gender) {
              handleSelectGender(data.value);
            }
          }}
        />
      </Box>
      <Box width={200}>
        <Button color="secondary" variant="outline" onClick={handleResetFilter}>
          Reset Filter
        </Button>
      </Box>
    </Box>
  );
};

export default Filter;
