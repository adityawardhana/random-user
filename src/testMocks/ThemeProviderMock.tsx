import React from "react";
import theme, { GlobalStyles } from "@adityawardhana/ts-react-ui/lib/theme";
import { ThemeProvider } from "styled-components";

export const setupTheme = (children: React.ReactNode) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};
