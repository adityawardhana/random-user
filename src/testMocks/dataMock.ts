export const resultAPIMock = {
  login: {
    username: "adityawrdhn",
  },
  name: {
    first: "adit",
    last: "w",
  },
  email: "adityawrdhn@gmail.com",
  gender: "male",
  registered: { date: "2003-08-26T21:56:32.575Z" },
};

export const usersMock = {
  params: { sortBy: "email", sortOrder: "ascend", page: 1, gender: "" },
  loading: false,
  users: [...Array(3)].map(() => ({ ...resultAPIMock })),
  keyword: "xxx",
  handleChangeSort: jest.fn(),
  handleChangePage: jest.fn(),
  handleChangeSearch: jest.fn(),
  setSearchFocus: jest.fn(),
  handleSelectGender: jest.fn(),
  handleResetFilter: jest.fn(),
};
