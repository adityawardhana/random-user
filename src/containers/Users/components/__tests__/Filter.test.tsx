/* eslint-disable testing-library/no-container */
/* eslint-disable testing-library/no-node-access */
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { usersMock } from "../../../../testMocks/dataMock";
import { setupTheme } from "../../../../testMocks/ThemeProviderMock";
import Provider from "../../Provider";
import useUsersMock from "../../useUser";
import Filter from "../Filter";

jest.mock("../../useUser", () => ({
  __esModule: true,
  default: jest.fn(),
}));

describe("Filter unit tests", () => {
  beforeEach(() => {
    (useUsersMock as any).mockReturnValue({ ...usersMock });
  });
  test("should render Filter", () => {
    const { container } = render(
      setupTheme(
        <Provider>
          <Filter />
        </Provider>
      )
    );
    expect(container).toMatchSnapshot();
  });

  test("should focus input", () => {
    const { container } = render(
      setupTheme(
        <Provider>
          <Filter />
        </Provider>
      )
    );

    fireEvent.focus(container?.getElementsByTagName("input")[0]);
    expect(usersMock.setSearchFocus).toHaveBeenCalled();
  });

  test("should open dropdown and select gender", async () => {
    const { container } = render(
      setupTheme(
        <Provider>
          <Filter />
        </Provider>
      )
    );

    fireEvent.click(container?.getElementsByTagName("button")[0]);
    await waitFor(() => {
      expect(screen.getAllByText(/Male/i)).toHaveLength(2)
    });
    fireEvent.click(screen.getAllByText(/Male/i)[0])
    expect(usersMock.handleSelectGender).toHaveBeenCalled()
  });

  test("should enable to click reset button", async () => {
    const { container } = render(
      setupTheme(
        <Provider>
          <Filter />
        </Provider>
      )
    );

    fireEvent.click(container?.getElementsByTagName("button")[1]);
    expect(usersMock.handleResetFilter).toHaveBeenCalled()
  });
});
