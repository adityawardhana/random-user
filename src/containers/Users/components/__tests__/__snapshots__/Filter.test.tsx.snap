// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter unit tests should not render Filter 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 40px;
}

.c1 {
  width: 200px;
  margin-right: 16px;
}

.c6 {
  position: relative;
}

.c9 {
  width: 200px;
}

.c8 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c5 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 16px;
  height: 16px;
  pointer-events: none;
}

.c5[type="button"] .c4 {
  cursor: pointer;
}

.c5 svg {
  width: 16px;
  height: 16px;
  fill: #152E3B;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
  border: 1px solid transparent;
  border-radius: 8px;
  border-color: transparent;
  background-color: #F4F5F8;
  color: #152E3B;
  padding: 12px 16px;
  height: 40px;
  width: auto;
  font-family: inherit;
  font-size: 14px;
  font-weight: 500;
  text-align: center;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  box-sizing: border-box;
  -webkit-transition: -webkit-transform 0.1s ease 0s,background 0.3s ease !important;
  -webkit-transition: transform 0.1s ease 0s,background 0.3s ease !important;
  transition: transform 0.1s ease 0s,background 0.3s ease !important;
  width: 100%;
}

.c7::before {
  content: "";
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c7:hover::before {
  background: rgba(255,255,255,0.1);
}

.c7:focus {
  outline: none;
}

.c7:focus-visible::before {
  background: rgba(50,50,50,0.05);
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
}

.c7:active {
  outline: none;
  -webkit-transform: scale(0.96) !important;
  -ms-transform: scale(0.96) !important;
  transform: scale(0.96) !important;
}

.c7:active::before {
  background: rgba(50,50,50,0.05);
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c7:disabled {
  color: #CDD0D7;
  cursor: default;
}

.c7:link,
.c7:visited,
.c7:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7[type="button"],
.c7[type="reset"],
.c7[type="submit"] {
  -webkit-appearance: none;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
  border: 1px solid transparent;
  border-radius: 8px;
  border-color: #152E3B;
  background-color: transparent;
  color: #152E3B;
  padding: 12px 16px;
  height: 40px;
  width: auto;
  font-family: inherit;
  font-size: 14px;
  font-weight: 500;
  text-align: center;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  box-sizing: border-box;
  -webkit-transition: -webkit-transform 0.1s ease 0s,background 0.3s ease !important;
  -webkit-transition: transform 0.1s ease 0s,background 0.3s ease !important;
  transition: transform 0.1s ease 0s,background 0.3s ease !important;
}

.c10::before {
  content: "";
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c10:hover::before {
  background: rgba(255,255,255,0.1);
}

.c10:focus {
  outline: none;
}

.c10:focus-visible::before {
  background: rgba(50,50,50,0.05);
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
}

.c10:active {
  outline: none;
  -webkit-transform: scale(0.96) !important;
  -ms-transform: scale(0.96) !important;
  transform: scale(0.96) !important;
}

.c10:active::before {
  background: rgba(50,50,50,0.05);
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c10:disabled {
  border-color: #CDD0D7;
  color: #CDD0D7;
  cursor: default;
}

.c10:link,
.c10:visited,
.c10:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10[type="button"],
.c10[type="reset"],
.c10[type="submit"] {
  -webkit-appearance: none;
}

.c2 {
  width: 100%;
  box-sizing: border-box;
  position: relative;
  display: block;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c2 .icon-search {
  top: 12px;
  right: 12px;
  position: absolute;
}

.c3 {
  box-shadow: none;
  margin-left: 0;
  height: 40px !important;
  border-color: transparent;
  background-color: #F4F5F8;
  color: #243E47;
  padding: 10px 34px 10px 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  font-size: 14px;
  line-height: 20px;
  border-radius: 8px;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      width="200"
    >
      <div
        class="c2"
      >
        <input
          class="c3"
          placeholder="Type your search here..."
          value="xxx"
        />
        <i
          class="c4 c5 icon-search"
          color="secondary"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-search"
            fill="currentColor"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M11.5 7a4.499 4.499 0 11-8.998 0A4.499 4.499 0 0111.5 7zm-.82 4.74a6 6 0 111.06-1.06l3.04 3.04a.75.75 0 11-1.06 1.06l-3.04-3.04z"
              fill-rule="evenodd"
            />
          </svg>
        </i>
      </div>
    </div>
    <div
      class="c1"
      width="200"
    >
      <div
        class="c6"
      >
        <button
          class="c7"
          color="secondary"
          shape="rectangle"
          style="justify-content: space-between;"
        >
          <div
            class="c8"
          >
            <s
              style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none; background-color: rgba(255, 255, 255, 0.4);"
            />
          </div>
          All
          <i
            class="c4 c5"
            color="secondary"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-chevron-down"
              fill="currentColor"
              height="16"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M12.78 6.22a.75.75 0 010 1.06l-4.25 4.25a.75.75 0 01-1.06 0L3.22 7.28a.75.75 0 011.06-1.06L8 9.94l3.72-3.72a.75.75 0 011.06 0z"
                fill-rule="evenodd"
              />
            </svg>
          </i>
        </button>
      </div>
    </div>
    <div
      class="c9"
      width="200"
    >
      <button
        class="c10"
        color="secondary"
        shape="rectangle"
      >
        <div
          class="c8"
        >
          <s
            style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none; background-color: rgba(255, 255, 255, 0.4);"
          />
        </div>
        Reset Filter
      </button>
    </div>
  </div>
</div>
`;

exports[`Filter unit tests should render Filter 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 40px;
}

.c1 {
  width: 200px;
  margin-right: 16px;
}

.c6 {
  position: relative;
}

.c9 {
  width: 200px;
}

.c8 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c5 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 16px;
  height: 16px;
  pointer-events: none;
}

.c5[type="button"] .c4 {
  cursor: pointer;
}

.c5 svg {
  width: 16px;
  height: 16px;
  fill: #152E3B;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
  border: 1px solid transparent;
  border-radius: 8px;
  border-color: transparent;
  background-color: #F4F5F8;
  color: #152E3B;
  padding: 12px 16px;
  height: 40px;
  width: auto;
  font-family: inherit;
  font-size: 14px;
  font-weight: 500;
  text-align: center;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  box-sizing: border-box;
  -webkit-transition: -webkit-transform 0.1s ease 0s,background 0.3s ease !important;
  -webkit-transition: transform 0.1s ease 0s,background 0.3s ease !important;
  transition: transform 0.1s ease 0s,background 0.3s ease !important;
  width: 100%;
}

.c7::before {
  content: "";
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c7:hover::before {
  background: rgba(255,255,255,0.1);
}

.c7:focus {
  outline: none;
}

.c7:focus-visible::before {
  background: rgba(50,50,50,0.05);
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
}

.c7:active {
  outline: none;
  -webkit-transform: scale(0.96) !important;
  -ms-transform: scale(0.96) !important;
  transform: scale(0.96) !important;
}

.c7:active::before {
  background: rgba(50,50,50,0.05);
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c7:disabled {
  color: #CDD0D7;
  cursor: default;
}

.c7:link,
.c7:visited,
.c7:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7[type="button"],
.c7[type="reset"],
.c7[type="submit"] {
  -webkit-appearance: none;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
  border: 1px solid transparent;
  border-radius: 8px;
  border-color: #152E3B;
  background-color: transparent;
  color: #152E3B;
  padding: 12px 16px;
  height: 40px;
  width: auto;
  font-family: inherit;
  font-size: 14px;
  font-weight: 500;
  text-align: center;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  box-sizing: border-box;
  -webkit-transition: -webkit-transform 0.1s ease 0s,background 0.3s ease !important;
  -webkit-transition: transform 0.1s ease 0s,background 0.3s ease !important;
  transition: transform 0.1s ease 0s,background 0.3s ease !important;
}

.c10::before {
  content: "";
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c10:hover::before {
  background: rgba(255,255,255,0.1);
}

.c10:focus {
  outline: none;
}

.c10:focus-visible::before {
  background: rgba(50,50,50,0.05);
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
}

.c10:active {
  outline: none;
  -webkit-transform: scale(0.96) !important;
  -ms-transform: scale(0.96) !important;
  transform: scale(0.96) !important;
}

.c10:active::before {
  background: rgba(50,50,50,0.05);
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c10:disabled {
  border-color: #CDD0D7;
  color: #CDD0D7;
  cursor: default;
}

.c10:link,
.c10:visited,
.c10:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10[type="button"],
.c10[type="reset"],
.c10[type="submit"] {
  -webkit-appearance: none;
}

.c2 {
  width: 100%;
  box-sizing: border-box;
  position: relative;
  display: block;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c2 .icon-search {
  top: 12px;
  right: 12px;
  position: absolute;
}

.c3 {
  box-shadow: none;
  margin-left: 0;
  height: 40px !important;
  border-color: transparent;
  background-color: #F4F5F8;
  color: #243E47;
  padding: 10px 34px 10px 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  font-size: 14px;
  line-height: 20px;
  border-radius: 8px;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      width="200"
    >
      <div
        class="c2"
      >
        <input
          class="c3"
          placeholder="Type your search here..."
          value="xxx"
        />
        <i
          class="c4 c5 icon-search"
          color="secondary"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-search"
            fill="currentColor"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M11.5 7a4.499 4.499 0 11-8.998 0A4.499 4.499 0 0111.5 7zm-.82 4.74a6 6 0 111.06-1.06l3.04 3.04a.75.75 0 11-1.06 1.06l-3.04-3.04z"
              fill-rule="evenodd"
            />
          </svg>
        </i>
      </div>
    </div>
    <div
      class="c1"
      width="200"
    >
      <div
        class="c6"
      >
        <button
          class="c7"
          color="secondary"
          shape="rectangle"
          style="justify-content: space-between;"
        >
          <div
            class="c8"
          >
            <s
              style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none; background-color: rgba(255, 255, 255, 0.4);"
            />
          </div>
          All
          <i
            class="c4 c5"
            color="secondary"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-chevron-down"
              fill="currentColor"
              height="16"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M12.78 6.22a.75.75 0 010 1.06l-4.25 4.25a.75.75 0 01-1.06 0L3.22 7.28a.75.75 0 011.06-1.06L8 9.94l3.72-3.72a.75.75 0 011.06 0z"
                fill-rule="evenodd"
              />
            </svg>
          </i>
        </button>
      </div>
    </div>
    <div
      class="c9"
      width="200"
    >
      <button
        class="c10"
        color="secondary"
        shape="rectangle"
      >
        <div
          class="c8"
        >
          <s
            style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none; background-color: rgba(255, 255, 255, 0.4);"
          />
        </div>
        Reset Filter
      </button>
    </div>
  </div>
</div>
`;
